### PR TITLE
[MM-18618] mac installer including signing, refactor pipeline

### PR DESCRIPTION
**Pull request by Willyfrog**
Before submitting, please confirm you've
 - [X] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [X] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [X] executed `npm run lint:js` for proper code formatting

Please provide the following information:

**Summary**

This reworked pipeline, builds depending if it's a regular PR or a branch planned for RC/final. Creates installers for windows and mac, and mac is also signed if the user has access to the context with the security info.

**Issue link**

[MM-18618](https://mattermost.atlassian.net/browse/MM-18618)

**Test Cases**

None needed

**Additional Notes**

it has merged master and previous PR such as https://github.com/mattermost/desktop/pull/1034 and https://github.com/mattermost/desktop/pull/1030

when the pipelines end, all the artifacts are stored together for moving them to S3 or access

Screenshot 2019-09-19 at 15 24 28

build of the 3 apps is done in parallel, as msi takes the longest
Screenshot 2019-09-19 at 15 23 05

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -67,19 +158,149 @@ jobs:
...
+            - build-linux
+            - build-win-no-installer
+            - build-mac-no-dmg
+          filters:
+            branches:
+              ignore:
+                - /^(v|V)\.\d+\.\d+(\.\d+)?(-(rc|RC)-.*/)?/
+                # todo: once working remove this one
+                - mac_build
+      - store:
```
Comment (cpanato) [inline .circleci/config.yml]: can you rename this to another better name? store what?

File: .circleci/config.yml [hunk 2]
```diff
@@ -67,19 +158,149 @@ jobs:
...
+          requires:
+            - check
+          context: codesign-certificates
+          filters:
+            branches:
+              only:
+                - /^(v|V)\.\d+\.\d+(\.\d+)?(-(rc|RC)-.*/)?/
+                # todo: once working remove this one
+                - mac_build
+      - store:
```
Comment (cpanato) [inline .circleci/config.yml]: same here

File: .circleci/config.yml (hunk 1 above)
Comment (wget) [reply .circleci/config.yml]: +1

File: .circleci/config.yml (hunk 2 above)
Comment (wget) [reply .circleci/config.yml]: +1

File: .circleci/config.yml (hunk 1 above)
Comment (Willyfrog) [reply .circleci/config.yml]: would `store_artifacts` be good enough? currently this job gets results from the rest of the jobs and publish them on the artifacts section

**Comment by Willyfrog** [comment]
already merged on #1044

**Pull request closed without merging**